### PR TITLE
EVALG-46: Create last value cache example for python and C++

### PR DESCRIPTION
**Pull request by albertoroblesh**
### Summary

Implement examples for last value cache in C++11 and Python

### Details and comments

I will update documentation once this PR is merged and everything is tied up

### CODE REVIEW

File: tutorials/last-value_cache/c++11/USER_QOS_PROFILES.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ of any type, including any warranty for fitness for any purpose. RTI is under
+ no obligation to maintain or support the Software.  RTI shall not be liable for
+ any incidental or consequential damages arising out of the use or inability to
+ use the software.
+ -->
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://community.rti.com/schema/current/rti_dds_profiles.xsd">
+    <qos_library name="MyLibrary">
+        <qos_profile name="MyProfile" is_default_qos="true">
+            <base_name>
+                <element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>
```
Comment (albertorobles2000) [inline tutorials/last-value_cache/c++11/USER_QOS_PROFILES.xml:16]: Should I comment this, use `<element>BuiltinQosLib::Generic.StrictReliable</element>` and add a comment that explains that if you don't want your data to be volatile you have to uncomment `<element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>` and comment  `<element>BuiltinQosLib::Generic.StrictReliable</element>`? Or should we use  `<element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>` out of the box for the example?

File: tutorials/last-value_cache/c++11/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+connextdds_add_application(
+    TARGET "subscriber"
+    LANG "C++11"
+    OUTPUT_NAME "home_automation_subscriber"
+    SOURCES
+        $<TARGET_OBJECTS:home_automation_CXX11_obj>
+        "${CMAKE_CURRENT_SOURCE_DIR}/home_automation_subscriber.cxx"
+    DEPENDENCIES
+    ${_CONNEXT_DEPENDENCIES}
+)
```
Comment (albertorobles2000) [inline tutorials/last-value_cache/c++11/CMakeLists.txt:50]: Regarding the section `3. Increase the history` should I add something to this example or do we leave it like that until we remove the `Under construction`

File: tutorials/last_value_cache/c++11/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+connextdds_add_application(
+    TARGET "subscriber"
+    LANG "C++11"
+    OUTPUT_NAME "home_automation_subscriber"
+    SOURCES
+        $<TARGET_OBJECTS:home_automation_CXX11_obj>
+        "${CMAKE_CURRENT_SOURCE_DIR}/home_automation_subscriber.cxx"
+    DEPENDENCIES
+    ${_CONNEXT_DEPENDENCIES}
+)
```
Comment (albertorobles2000) [inline tutorials/last_value_cache/c++11/CMakeLists.txt:50]: Should I comment this, use `<element>BuiltinQosLib::Generic.StrictReliable</element>` and add a comment that explains that if you don't want your data to be volatile you have to uncomment `<element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>` and comment  `<element>BuiltinQosLib::Generic.StrictReliable</element>`? Or should we use  `<element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>` out of the box for the example?

File: tutorials/last_value_cache/c++11/USER_QOS_PROFILES.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ of any type, including any warranty for fitness for any purpose. RTI is under
+ no obligation to maintain or support the Software.  RTI shall not be liable for
+ any incidental or consequential damages arising out of the use or inability to
+ use the software.
+ -->
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://community.rti.com/schema/current/rti_dds_profiles.xsd">
+    <qos_library name="MyLibrary">
+        <qos_profile name="MyProfile" is_default_qos="true">
+            <base_name>
+                <element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>
```
Comment (albertorobles2000) [inline tutorials/last_value_cache/c++11/USER_QOS_PROFILES.xml:16]: Should I comment this, use `<element>BuiltinQosLib::Generic.StrictReliable</element>` and add a comment that explains that if you don't want your data to be volatile you have to uncomment `<element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>` and comment  `<element>BuiltinQosLib::Generic.StrictReliable</element>`? Or should we use  `<element>BuiltinQosLib::Generic.KeepLastReliable.TransientLocal</element>` out of the box for the example?

File: tutorials/last_value_cache/c++11/home_automation_audited.cxx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+void publish_sensor(
+    const std::string& sensor_name,
+    const std::string& room_name)
+{
+    dds::domain::DomainParticipant participant(0);
+    dds::topic::Topic<DeviceStatus> topic(participant, "WindowStatus");
+    dds::pub::DataWriter<DeviceStatus> writer(
+            topic,
+            dds::core::QosProvider::Default().datawriter_qos("Last10Cache"));
```
Comment (albertoroblesh) [inline tutorials/last_value_cache/c++11/home_automation_audited.cxx]: Should we create a unique publisher with a history of 10? And just modify the subscriber/audit file, so we don't need to publishers?

File: tutorials/last_value_cache/c++11/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    LANG "C++11"
+    OUTPUT_NAME "home_automation_audit"
+    SOURCES
+        $<TARGET_OBJECTS:home_automation_CXX11_obj>
+        "${CMAKE_CURRENT_SOURCE_DIR}/home_automation_audit.cxx"
+    DEPENDENCIES
+    ${_CONNEXT_DEPENDENCIES}
+)
+
+connextdds_add_application(
```
Comment (albertoroblesh) [inline tutorials/last_value_cache/c++11/CMakeLists.txt]: https://github.com/alexcamposruiz/rticonnextdds-examples/pull/2/files#r1615733448

File: tutorials/last_value_cache/c++11/home_automation_audited.cxx (hunk 5 above)
Comment (alexcamposruiz) [reply tutorials/last_value_cache/c++11/home_automation_audited.cxx]: good question - I don't like having two publishers, but I also don't want to distract from the first part of the module by always setting the history to 10.

🤔

Comment (alexcamposruiz) [reply tutorials/last_value_cache/c++11/home_automation_audited.cxx]: I think we should update the instructions in the guide to ask the user to manually modify the publisher application to use the new profile when a publisher needs to be audited, and delete the extra publisher file from this PR.

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]
Delete the audited publisher app (see other comment)

File: tutorials/last_value_cache/c++11/home_automation_audited.cxx (hunk 5 above)
Comment (albertoroblesh) [reply tutorials/last_value_cache/c++11/home_automation_audited.cxx]: Sounds good to me :+1:

**Review by alexcamposruiz — APPROVED** [review]

**Pull request merged**